File: app/src/main/java/net/harimurti/tv/extra/Network.java
package net.harimurti.tv.extra;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

public class Network {
    public static boolean IsConnected(Context context) {
        ConnectivityManager manager = (ConnectivityManager)context
                .getSystemService(Context.CONNECTIVITY_SERVICE);

        if (manager == null) return false;

        NetworkInfo network = manager.getActiveNetworkInfo();
        if (network == null) return false;

        return network.isConnected();
    }
}

File: app/src/main/java/net/harimurti/tv/extra/Network.kt
package net.harimurti.tv.extra

import android.content.Context
import android.net.ConnectivityManager
import android.os.Build
import net.harimurti.tv.App

@Suppress("DEPRECATION")
class Network {
    private val context = App.context
    fun isConnected(): Boolean {
        val cm = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
            cm.getNetworkCapabilities(cm.activeNetwork) != null
        } else {
            cm.activeNetworkInfo?.isConnected ?: false
        }
    }
}